arkouda/dataframe.py

```diff
@@ -5326,6 +5609,322 @@ def from_return_msg(cls, rep_msg):
         return cls(columns, idx)
 
 
+class _LocIndexer:
+    def __init__(self, df):
+        self.df = df
+
+    def __getitem__(self, key):
+        if isinstance(key, tuple) and len(key) == 2:
+            return self._get_row_col(key[0], key[1])
+        if isinstance(key, list):
+            key = array(key)
+        if isinstance(key, Series):
+            key = key.values
+        if is_supported_scalar(key) and self.df.index.dtype == dtype(type(key)):
+            return self.df._get_rows(indexof1d(array([key]), self.df.index.values))
+
+        if isinstance(key, pdarray) and key.dtype == self.df.index.dtype:
+            return self.df._get_rows(indexof1d(key, self.df.index.values))
```

Comment:
do we expect there to be multiple occurrences of `key`/the values of `key` in `self.df.index.values`? if not then it would be more efficient to use `ak.find` rather than `indexof1d`